### PR TITLE
multi monitor support

### PR DESCRIPTION
**Pull request by gaamdalurt**
Hi, I added multi monitor support to Fly-Pie.
It was done by setting the background x and y to the current monitor x and y, then keeping everything relative to the background.

By the way, only the current monitor is used. If you wish to accomplish a ore complicated effect like using the merged surface of all monitors let me know! :-)
(It'll be fantastic to have such an option into fly pie, choosing to use only the current monitor or all the monitor together)

### CODE REVIEW

**Comment by Schneegans** [comment]
Thank you so much for this contribution. I will get an additional monitor later this week so that I can test this. I think for Fly-Pie 3, it will be sufficient to use the current monitor only. Allowing gestures over multiple monitors would be nice but if it would increase the code complexity significantly we should re-think whether it's going to be a common use case...

**Comment by gaamdalurt** [comment]
Perfect, I'll wait for your tests, count me in if you need any help. :-)

File: daemon/Background.js [hunk 1]
```diff
@@ -65,7 +65,8 @@ class Background extends Clutter.Actor {
     // Switch monitor side when the preview-on-right-side settings key changes.
     this._settings.connect('changed::preview-on-right-side', () => {
       if (this._previewMode) {
-        this.x = this._settings.get_boolean('preview-on-right-side') ? this.width : 0;
+        // Set x accounting monitor x as a starting point
+        this.x = this._settings.get_boolean('preview-on-right-side') ? this.width + Main.layoutManager.currentMonitor.x : Main.layoutManager.currentMonitor.x;
```
Comment (Schneegans) [inline daemon/Background.js:69]: Maybe this long line would be a bit more readable like this?

```suggestion
        this.x = Main.layoutManager.currentMonitor.x;
        if (this._settings.get_boolean('preview-on-right-side')) {
          this.x += this.width;
        }
```

File: daemon/Background.js [hunk 2]
```diff
@@ -111,14 +112,15 @@ class Background extends Clutter.Actor {
 
       // Set background size to one half of the monitor.
       this.width = Main.layoutManager.currentMonitor.width / 2;
-      this.x     = this._settings.get_boolean('preview-on-right-side') ? this.width : 0;
-
+      this.height = Main.layoutManager.currentMonitor.height;
+      // Set x accounting monitor x as a starting point
+      this.x     = this._settings.get_boolean('preview-on-right-side') ? this.width + Main.layoutManager.currentMonitor.x : Main.layoutManager.currentMonitor.x;
```
Comment (Schneegans) [inline daemon/Background.js:117]: Here as well:

```suggestion
      this.x = Main.layoutManager.currentMonitor.x;
      if (this._settings.get_boolean('preview-on-right-side')) {
        this.x += this.width;
      }
```

**Review by Schneegans — CHANGES_REQUESTED** [review]
Works like a charm! Could you run the `./clang-format.sh` script on your end to format the code properly? Then I will happily merge this.

**Regarding more advanced multi-monitor support:** In my opinion it's sufficient to have the menus only on the current monitor. However it feels a bit weird to click on the content of the other monitor while a menu is opened and nothing happens. What can we do about this?
* Cancel the menu when the user clicks on another monitor: I would be against this as it would force the user to pay attention not to move the mouse too far.
* Darken the other monitors as well: Maybe a better solution but still a little surprising that nothing happens.
* Constrain the pointer movement to the current monitor: Would be a solid solution, but I am not sure if that is possible on Wayland...?
* Go back to your initial suggestion and make Fly-Pie work across multiple screens. If you feel adventurous, you could try to implement this. But I fear that it makes the code way more complicated and I am not sure that it's worth the effort...

What do you think? Maybe we should try the pointer-constrain solution?

**Comment by gaamdalurt** [comment]
It should be properly formatted now.
As you said it feels weird to click on the other monitor while the menu is opened.
Pointer constraint could be a good solution even if I feel like it is something unexpected from a user point of view, probably shade all the screens could be a better solution.
It's obvious that if the code base becomes a lot complicated we will revert to the constraint solution, but if I understood well how GNOME treat monitors, probably it will end up being not so complicated.

PS. Great extension BTW, really useful on a daily basis.
PPS. Excuse for some English mistakes, it's not my mother tongue and I'm learning just a little bit more everyday.

**Comment by Schneegans** [comment]
Thank you once more! I am looking forward to future contributions from your side!

PS. You're welcome :smiley:

**Pull request merged**